### PR TITLE
Loading link provider only when the user opens the menu tab or change its position

### PR DESCRIPTION
**Pull request by YaroslavOvdii**
@sofiiakvasnevska 

## Issue
https://github.com/Fliplet/fliplet-studio/issues/5479

## Description
Loading link provider only when the user opens the menu tab or change its position

## Screenshots/screencasts
https://share.getcloudapp.com/RBudz2z9

## Backward compatibility

This change is fully backward compatible.

## Notes
In this solution, I've to take note of an issue my previous solution cased,  so now the user will not lose his menu items on saving while he wasn't open menu tabs.

### CODE REVIEW

**Comment by tonytlwu** [comment]
@squallstar Could you take a look at this one? I know you were assessing another issue with missing menu items the other day and found that the previous fix for this issue was creating problems. You might be a better person for reviewing the new fix too.

File: js/interface.js [hunk 1]
```diff
@@ -441,6 +432,52 @@
     });
   }
 
+  /**
+   * Function that deside should we run intiLinkProvider function or not
+   * @param {int} menuItemId - an id of the menu item we should check
+   * @returns {void}
+  */
+  function checkIsProviderInited(menuItemId) {
```
Comment (squallstar) [inline js/interface.js]: let's rename to `checkIfLinkProviderInitialized`

File: js/interface.js [hunk 2]
```diff
@@ -441,6 +432,52 @@
     });
   }
 
+  /**
+   * Function that deside should we run intiLinkProvider function or not
```
Comment (squallstar) [inline js/interface.js]: > This function ensures the initLinkProvider method has been initialised

File: js/interface.js (hunk 1 above)
Comment (YaroslavOvdii) [reply js/interface.js]: @squallstar renamed

File: js/interface.js (hunk 2 above)
Comment (YaroslavOvdii) [reply js/interface.js]: @squallstar changed description

**Pull request merged**